drivers/cleanup

```diff
@@ -0,0 +1,137 @@
+                     max_age hours
+    -a --abort       abort any currently running operation (GC or coalesce)
+    -q --query       query the current state (GC'ing, coalescing or not running)
+    -x --disable     disable GC/coalesce (will be in effect until you exit)
+    -t --debug       see Debug below
+
+Options:
+    -b --background  run in background (return immediately) (valid for -g only)
+    -f --force       continue in the presence of VHDs with errors (when doing
+                     GC, this might cause removal of any such VHDs) (only valid
+                     for -G) (DANGEROUS)
+
+Debug:
+    The --debug parameter enables manipulation of LVHD VDIs for debugging
+    purposes.  ** NEVER USE IT ON A LIVE VM **
+    The following parameters are required:
+    -t --debug <cmd> <cmd> is one of "activate", "deactivate", "inflate",
+                     "deflate".
+    -v --vdi_uuid    VDI UUID
+    """
+    #-d --dry-run     don't actually perform any SR-modifying operations
+    print(output)
+    cleanup.Util.log("(Invalid usage)")
+    sys.exit(1)
+
+##############################################################################
+#
+#  CLI
+#
+def main():
+    action = ""
+    uuid = ""
+    background = False
+    force = False
+    dryRun = False
+    maxAge = 0
+    debug_cmd = ""
+    vdi_uuid = ""
+    shortArgs = "gGc:aqxu:bfdt:v:"
+    longArgs = ["gc", "gc_force", "clean_cache", "abort", "query", "disable",
+            "uuid=", "background", "force", "dry-run", "debug=", "vdi_uuid="]
+
+    try:
+        opts, args = getopt.getopt(sys.argv[1:], shortArgs, longArgs)
+    except getopt.GetoptError:
+        usage()
+    for o, a in opts:
+        if o in ("-g", "--gc"):
+            action = "gc"
+        if o in ("-G", "--gc_force"):
+            action = "gc_force"
+        if o in ("-c", "--clean_cache"):
+            action = "clean_cache"
+            maxAge = int(a)
+        if o in ("-a", "--abort"):
+            action = "abort"
+        if o in ("-q", "--query"):
+            action = "query"
+        if o in ("-x", "--disable"):
+            action = "disable"
+        if o in ("-u", "--uuid"):
+            uuid = a
+        if o in ("-b", "--background"):
+            background = True
+        if o in ("-f", "--force"):
+            force = True
+        if o in ("-d", "--dry-run"):
+            cleanup.Util.log("Dry run mode")
+            dryRun = True
+        if o in ("-t", "--debug"):
+            action = "debug"
+            debug_cmd = a
+        if o in ("-v", "--vdi_uuid"):
+            vdi_uuid = a
+
+    if not action or not uuid:
+        usage()
+    if action == "debug" and not (debug_cmd and vdi_uuid) or \
+            action != "debug" and (debug_cmd or vdi_uuid):
+        usage()
+
+    if action != "query" and action != "debug":
+        print("All output goes to log")
+
+    if action == "gc":
+        cleanup.gc(None, uuid, background, dryRun)
```

Comment:
need a `from sm import cleanup`?

Reply:
There is one below, which will always have executed by the time this is called, but I'll move it up to the top.